walk-in-interview/src/main/java/com/google/utils/ServletUtils.java

```diff
@@ -0,0 +1,142 @@
+        }
+        return value;
+    }
+
+    /**
+     * Gets the integer parameter from html form input.
+     *
+     * @param request Http request.
+     * @param parameterName Input id in html element.
+     * @param defaultValue Default integer.
+     * @return Parsed integer or default value if exception occur
+     */
+    public static int getIntParameter(HttpServletRequest request, String parameterName, int defaultValue) {
+        String resultStr = request.getParameter(parameterName).trim();
+
+        try {
+            int result = Integer.parseInt(resultStr);
+
+            return result;
+        } catch (NumberFormatException e) {
+            // TODO(issue/12): error handling; can consider to add logging to log the error
+
+            return defaultValue;
+        }
+    }
+
+    /**
+     * Gets the long parameter from html form input.
+     *
+     * @param request Http request.
+     * @param parameterName Input id in html element.
+     * @param defaultValue Default integer.
+     * @return Parsed integer or default value if exception occur
+     */
+    public static long getLongParameter(HttpServletRequest request, String parameterName, long defaultValue) {
+        String resultStr = request.getParameter(parameterName).trim();
+
+        try {
+            long result = Integer.parseInt(resultStr);
+
+            return result;
+        } catch (NumberFormatException e) {
+            // TODO(issue/12): error handling; can consider to add logging to log the error
+
+            return defaultValue;
+        }
+    }
+
+    /**
+     * Gets the float parameter from html form input.
+     *
+     * @param request Http request.
+     * @param parameterName Input id in html element.
+     * @param defaultValue Default integer.
+     * @return Parsed float or default value if exception occur
+     */
+    public static float getFloatParameter(HttpServletRequest request, String parameterName, float defaultValue) {
+        String resultStr = request.getParameter(parameterName).trim();
+
+        try {
+            float result = Float.parseFloat(resultStr);
+
+            return result;
+        } catch (NumberFormatException e) {
+            // TODO(issue/12): error handling; can consider to add logging to log the error
+
+            return defaultValue;
+        }
+    }
+
+    /**
+     * Gets the LocalDate parameter from html form input.
+     * @param request Http request.
+     * @param parameterName Input id in html element.
+     * @param defaultValue Default integer.
+     * @return Parsed date or default value if exception occur
+     */
+    public static Date getDateParameter(HttpServletRequest request,
+                                        String parameterName, Date defaultValue) {
+        String resultStr = request.getParameter(parameterName).trim();
+
+        try {
+            String pattern = "yyyy-MM-dd";
+            Date result = new SimpleDateFormat(pattern).parse(resultStr);
```

Comment:
Does front end and backend agree on the date format?
Maybe can look at ISO date format.

Reply:
We added into the design doc, yyyy-mm-dd